scripts/bootloader/provision.py

```diff
@@ -39,7 +48,7 @@ def add_hw_counters(provision_data, num_counter_slots_version, mcuboot_counters_
     assert num_counter_slots_version % 2 == 0, "--num-counters-slots-version must be an even number"
     assert mcuboot_counters_slots    % 2 == 0, "--mcuboot-counters-slots     must be an even number"
 
-    provision_data += struct.pack('H', 1) # Type "counter collection"
+    provision_data += struct.pack('H', BL_COLLECTION_TYPE_MONOTONIC_COUNTERS)
```

Comment:
This line is fine but I think that this function will not work for nRF54L15.  The counters for this have 32 bit length but here in python when we add the counters in line 57 we do:
`provision_data += bytes(2 * num_counter_slots_version * [0xFF])`

Which should be wrong I think, we need to update the function.